File: SubControls/SelectLocation.ascx.vb
'
' DotNetNuke® - http://www.dnnsoftware.com
' Copyright (c) 2002-2014
' by DNNCorp
'
' Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
' documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
' the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
' to permit persons to whom the Software is furnished to do so, subject to the following conditions:
'
' The above copyright notice and this permission notice shall be included in all copies or substantial portions 
' of the Software.
'
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
' THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
' CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
' DEALINGS IN THE SOFTWARE.
'
Imports System
Imports System.Web.UI.WebControls
Imports DotNetNuke.Security

Namespace DotNetNuke.Modules.Events

    Partial Class SelectLocation
        Inherits EventBase

#Region "Properties"

        Private _selectedLocation As New ArrayList
        Private _gotLocations As Boolean = False
        Private _modulelocationids As New ArrayList
        Public Property SelectedLocation() As ArrayList
            Get
                'have selected the location before?
                If Not _gotLocations Then
                    _gotLocations = True
                    _selectedLocation.Clear()

                    'is there a default module location when location select has been disabled
                    'if not has it been passed in as a parameter 
                    'if not is there a default module location when location select has not been disabled
                    'if not is there as setting in cookies available?
                    If Settings.Enablelocations = EventModuleSettings.DisplayLocations.DoNotDisplay Then
                        If Settings.ModuleLocationsSelected = EventModuleSettings.LocationsSelected.All Then
                            _selectedLocation.Clear()
                            _selectedLocation.Add("-1")
                        Else
                            _selectedLocation.Clear()
                            For Each location As Integer In Settings.ModuleLocationIDs
                                _selectedLocation.Add(location)
                            Next
                        End If
                    ElseIf Not Request.Params("Location") = Nothing Then
                        Dim objSecurity As New PortalSecurity
                        Dim tmpLocation As String = Request.Params("Location")
                        tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoScripting)
                        tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoSQL)
                        Dim oCntrlEventLocation As New EventLocationController
                        Dim oEventLocation As EventLocationInfo = oCntrlEventLocation.EventsLocationGetByName(tmpLocation, PortalSettings.PortalId)
                        If Not oEventLocation Is Nothing Then
                            _selectedLocation.Add(oEventLocation.Location)
                        End If
                    ElseIf Settings.ModuleLocationsSelected <> EventModuleSettings.LocationsSelected.All Then
                        _selectedLocation.Clear()
                        For Each location As Integer In Settings.ModuleLocationIDs
                            _selectedLocation.Add(location)
                        Next
                    ElseIf Request.Cookies("DNNEvents") Is Nothing Then
                        _selectedLocation.Clear()
                        _selectedLocation.Add("-1")
                    Else
                        'Do we have a special one for this module?
                        If Request.Cookies("DNNEvents")("EventLocation" & ModuleId) Is Nothing Then
                            _selectedLocation.Clear()
                            _selectedLocation.Add("-1")
                        Else
                            'Yes there is one!
                            Dim objSecurity As New PortalSecurity
                            Dim tmpLocation As String = Request.Cookies("DNNEvents")("EventLocation" & ModuleId)
                            tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoScripting)
                            tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoSQL)
                            Dim tmpArray() As String = Split(tmpLocation, ",")
                            For i As Integer = 0 To tmpArray.Length - 1
                                If tmpArray(i) <> "" Then
                                    _selectedLocation.Add(CInt(tmpArray(i)))
                                End If
                            Next
                        End If
                    End If
                End If
                Return _selectedLocation
            End Get
            Set(ByVal value As ArrayList)
                Try
                    _selectedLocation = value
                    _gotLocations = True
                    Response.Cookies("DNNEvents")("EventLocation" & ModuleId) = String.Join(",", CType(_selectedLocation.ToArray(GetType(String)), String()))
                    Response.Cookies("DNNEvents").Expires = DateTime.Now.AddMinutes(2)
                    Response.Cookies("DNNEvents").Path = "/"
                Catch ex As Exception
                End Try
            End Set
        End Property
        Public Property ModuleLocationIDs As ArrayList
            Get
                Return _modulelocationids
            End Get
            Set(value As ArrayList)
                _modulelocationids = value
            End Set
        End Property
#End Region

#Region "Public Methods"
        Public Sub StoreLocations()
            SelectedLocation.Clear()
            Dim lstLocations As New ArrayList
            If Settings.Enablelocations = EventModuleSettings.DisplayLocations.SingleSelect Then
                lstLocations.Add(ddlLocations.SelectedValue)
            Else
                If ddlLocations.CheckedItems.Count <> ddlLocations.Items.Count Then
                    For Each item As Telerik.Web.UI.RadComboBoxItem In ddlLocations.CheckedItems
                        lstLocations.Add(item.Value)
                    Next
                Else
                    lstLocations.Add("-1")
                End If
            End If
            SelectedLocation = lstLocations
        End Sub

#End Region

#Region "Event Handlers"
        Private Sub Page_Load(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.Load
            Try
                ' Add the external Validation.js to the Page
                Const csname As String = "ExtValidationScriptFile"
                Dim cstype As Type = Reflection.MethodBase.GetCurrentMethod().GetType()
                Dim cstext As String = "<script src=""" & ResolveUrl("~/DesktopModules/Events/Scripts/Validation.js") & """ type=""text/javascript""></script>"
                If Not Page.ClientScript.IsClientScriptBlockRegistered(csname) Then
                    Page.ClientScript.RegisterClientScriptBlock(cstype, csname, cstext, False)
                End If

                ddlLocations.EmptyMessage = Localization.GetString("NoLocations", LocalResourceFile)
                ddlLocations.Localization.AllItemsCheckedString = Localization.GetString("AllLocations", LocalResourceFile)
                ddlLocations.Localization.CheckAllString = Localization.GetString("SelectAllLocations", LocalResourceFile)
                If Settings.Enablelocations = EventModuleSettings.DisplayLocations.SingleSelect Then
                    ddlLocations.CheckBoxes = False
                End If

                If Not Page.IsPostBack Then
                    'Bind DDL
                    Dim ctrlEventLocations As New EventLocationController
                    Dim lstLocations As ArrayList = ctrlEventLocations.EventsLocationList(PortalId)

                    Dim arrLocations As New ArrayList
                    If Settings.Restrictlocations Then
                        For Each dbLocation As EventLocationInfo In lstLocations
                            For Each location As Integer In Settings.ModuleLocationIDs
                                If dbLocation.Location = location Then
                                    arrLocations.Add(dbLocation)
                                End If
                            Next
                        Next
                    Else
                        arrLocations.AddRange(lstLocations)
                    End If

                    If lstLocations.Count = 0 Then
                        Visible = False
                        SelectedLocation.Clear()
                        Exit Sub
                    End If

                    'Restrict locations by events in time frame.
                    If Settings.RestrictLocationsToTimeFrame Then
                        'Only for list view.
                        Dim whichView As String = String.Empty
                        If Not Request.QueryString("mctl") = Nothing And ModuleId = CType(Request.QueryString("ModuleID"), Integer) Then
                            If Request("mctl").EndsWith(".ascx") Then
                                whichView = Request("mctl")
                            Else
                                whichView = Request("mctl") & ".ascx"
                            End If
                        End If
                        If whichView.Length = 0 Then
                            If Not Request.Cookies.Get("DNNEvents" & ModuleId) Is Nothing Then
                                whichView = Request.Cookies.Get("DNNEvents" & ModuleId).Value
                            Else
                                whichView = Settings.DefaultView
                            End If
                        End If

                        If (whichView = "EventList.ascx" OrElse whichView = "EventRpt.ascx") Then
                            Dim objEventInfoHelper As New EventInfoHelper(ModuleId, TabId, PortalId, Settings)
                            Dim lstEvents As ArrayList

                            Dim getSubEvents As Boolean = Settings.MasterEvent
                            Dim numDays As Integer = Settings.EventsListEventDays
                            Dim displayDate, startDate, endDate As DateTime
                            If Settings.ListViewUseTime Then
                                displayDate = DisplayNow()
                            Else
                                displayDate = DisplayNow.Date
                            End If
                            If Settings.EventsListSelectType = "DAYS" Then
                                startDate = displayDate.AddDays(Settings.EventsListBeforeDays * -1)
                                endDate = displayDate.AddDays(Settings.EventsListAfterDays * 1)
                            Else
                                startDate = displayDate
                                endDate = displayDate.AddDays(numDays)
                            End If

                            lstEvents = objEventInfoHelper.GetEvents(startDate, endDate, getSubEvents, _
                                                                     New ArrayList({"-1"}), New ArrayList({"-1"}), -1, -1)

                            Dim eventLocationIds As ArrayList = New ArrayList()
                            For Each lstEvent As EventInfo In lstEvents
                                eventLocationIds.Add(lstEvent.Location)
                            Next
                            For Each lstLocation As EventLocationInfo In lstLocations
                                If Not eventLocationIds.Contains(lstLocation.Location) Then
                                    arrLocations.Remove(lstLocation)
                                End If
                            Next
                        End If
                    End If

                    'Bind locations.
                    ddlLocations.DataSource = arrLocations
                    ddlLocations.DataBind()

                    If Settings.Enablelocations = EventModuleSettings.DisplayLocations.SingleSelect Then
                        ddlLocations.Items.Insert(0, New Telerik.Web.UI.RadComboBoxItem(Localization.GetString("AllLocations", LocalResourceFile), "-1"))
                        ddlLocations.SelectedIndex = 0
                    End If
                    ddlLocations.OnClientDropDownClosed = "function() { btnUpdateClick('" + btnUpdate.UniqueID + "','" + ddlLocations.ClientID + "');}"
                    ddlLocations.OnClientLoad = "function() { storeText('" + ddlLocations.ClientID + "');}"
                    If Settings.Enablelocations = EventModuleSettings.DisplayLocations.SingleSelect Then
                        For Each location As Integer In SelectedLocation
                            ddlLocations.SelectedIndex = ddlLocations.FindItemByValue(location.ToString).Index
                            Exit For
                        Next
                    Else
                        For Each location As Integer In SelectedLocation
                            For Each item As Telerik.Web.UI.RadComboBoxItem In ddlLocations.Items
                                If item.Value = location.ToString Then
                                    item.Checked = True
                                End If
                            Next
                        Next

                        If CInt(SelectedLocation.Item(0)) = -1 Then
                            For Each item As Telerik.Web.UI.RadComboBoxItem In ddlLocations.Items
                                item.Checked = True
                            Next
                        End If
                    End If

                End If
            Catch exc As Exception
                'ProcessModuleLoadException(Me, exc)
            End Try
        End Sub

        Private Sub btnUpdate_Click(sender As Object, e As EventArgs) Handles btnUpdate.Click
            StoreLocations()

            ' Fire the LocationSelected event...
            Dim args As CommandEventArgs = New CommandEventArgs(SelectedLocation.ToString, Nothing)
            RaiseEvent LocationSelectedChanged(Me, args)
        End Sub

        Public Event LocationSelectedChanged(ByVal sender As Object, ByVal e As CommandEventArgs)

#End Region

#Region " Web Form Designer Generated Code "

        'This call is required by the Web Form Designer.
        <DebuggerStepThrough()> Private Sub InitializeComponent()

        End Sub

        Private Sub Page_Init(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.Init
            'CODEGEN: This method call is required by the Web Form Designer
            'Do not modify it using the code editor.
            InitializeComponent()
        End Sub
#End Region

    End Class

End Namespace

File: SubControls/SelectLocation.ascx.cs
#region Copyright

// 
// DotNetNuke® - http://www.dotnetnuke.com
// Copyright (c) 2002-2018
// by DotNetNuke Corporation
// 
// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated 
// documentation files (the "Software"), to deal in the Software without restriction, including without limitation 
// the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and 
// to permit persons to whom the Software is furnished to do so, subject to the following conditions:
// 
// The above copyright notice and this permission notice shall be included in all copies or substantial portions 
// of the Software.
// 
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
// THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
// CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
// DEALINGS IN THE SOFTWARE.
//

#endregion


namespace DotNetNuke.Modules.Events
{
    using System;
    using System.Collections;
    using System.Diagnostics;
    using System.Reflection;
    using System.Web.UI.WebControls;
    using Security;
    using Services.Localization;
    using global::Components;
    using Telerik.Web.UI;
    using EventInfo = global::Components.EventInfo;

    public partial class SelectLocation : EventBase
    {
        #region Public Methods

        public void StoreLocations()
        {
            SelectedLocation.Clear();
            var lstLocations = new ArrayList();
            if (Settings.Enablelocations == EventModuleSettings.DisplayLocations.SingleSelect)
            {
                lstLocations.Add(ddlLocations.SelectedValue);
            }
            else
            {
                if (ddlLocations.CheckedItems.Count > 0 && ddlLocations.CheckedItems.Count != ddlLocations.Items.Count)
                {
                    foreach (var item in ddlLocations.CheckedItems)
                    {
                        lstLocations.Add(item.Value);
                    }
                }
                else
                {
                    lstLocations.Add("-1");
                }
            }
            SelectedLocation = lstLocations;
        }

        #endregion

        #region Properties

        private ArrayList _selectedLocation = new ArrayList();
        private bool _gotLocations;

        public ArrayList SelectedLocation
        {
            get
                {
                    //have selected the location before
                    if (!_gotLocations)
                    {
                        _gotLocations = true;
                        _selectedLocation.Clear();

                        //is there a default module location when location select has been disabled
                        //if not has it been passed in as a parameter
                        //if not is there a default module location when location select has not been disabled
                        //if not is there as setting in cookies available
                        if (Settings.Enablelocations == EventModuleSettings.DisplayLocations.DoNotDisplay)
                        {
                            if (Settings.ModuleLocationsSelected == EventModuleSettings.LocationsSelected.All)
                            {
                                _selectedLocation.Clear();
                                _selectedLocation.Add("-1");
                            }
                            else
                            {
                                _selectedLocation.Clear();
                                foreach (var location in Settings.ModuleLocationIDs)
                                {
                                    _selectedLocation.Add(location);
                                }
                            }
                        }
                        else if (!(Request.Params["Location"] == null))
                        {
                            var objSecurity = new PortalSecurity();
                            var tmpLocation = Request.Params["Location"];
                            tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoScripting);
                            tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoSQL);
                            var oCntrlEventLocation = new EventLocationController();
                            var oEventLocation =
                                oCntrlEventLocation.EventsLocationGetByName(tmpLocation, PortalSettings.PortalId);
                            if (!ReferenceEquals(oEventLocation, null))
                            {
                                _selectedLocation.Add(oEventLocation.Location);
                            }
                        }
                        else if (Settings.ModuleLocationsSelected != EventModuleSettings.LocationsSelected.All)
                        {
                            _selectedLocation.Clear();
                            foreach (var location in Settings.ModuleLocationIDs)
                            {
                                _selectedLocation.Add(location);
                            }
                        }
                        else if (ReferenceEquals(Request.Cookies["DNNEvents"], null))
                        {
                            _selectedLocation.Clear();
                            _selectedLocation.Add("-1");
                        }
                        else
                        {
                            //Do we have a special one for this module
                            if (ReferenceEquals(
                                Request.Cookies["DNNEvents"]["EventLocation" + Convert.ToString(ModuleId)],
                                null))
                            {
                                _selectedLocation.Clear();
                                _selectedLocation.Add("-1");
                            }
                            else
                            {
                                //Yes there is one!
                                var objSecurity = new PortalSecurity();
                                var tmpLocation =
                                    Convert.ToString(
                                        Request.Cookies["DNNEvents"][
                                            "EventLocation" + Convert.ToString(ModuleId)]);
                                tmpLocation =
                                    objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoScripting);
                                tmpLocation = objSecurity.InputFilter(tmpLocation, PortalSecurity.FilterFlag.NoSQL);
                                var tmpArray = tmpLocation.Split(',');
                                for (var i = 0; i <= tmpArray.Length - 1; i++)
                                {
                                    if (tmpArray[i] != "")
                                    {
                                        _selectedLocation.Add(int.Parse(tmpArray[i]));
                                    }
                                }
                            }
                        }
                    }
                    return _selectedLocation;
                }
            set
                {
                    try
                    {
                        _selectedLocation = value;
                        _gotLocations = true;
                        Response.Cookies["DNNEvents"]["EventLocation" + Convert.ToString(ModuleId)] =
                            string.Join(",", (string[]) _selectedLocation.ToArray(typeof(string)));
                        Response.Cookies["DNNEvents"].Expires = DateTime.Now.AddMinutes(2);
                        Response.Cookies["DNNEvents"].Path = "/";
                    }
                    catch (Exception)
                    { }
                }
        }

        public ArrayList ModuleLocationIDs { get; set; } = new ArrayList();

        #endregion

        #region Event Handlers

        private void Page_Load(object sender, EventArgs e)
        {
            try
            {
                // Add the external Validation.js to the Page
                const string csname = "ExtValidationScriptFile";
                var cstype = MethodBase.GetCurrentMethod().GetType();
                var cstext = "<script src=\"" + ResolveUrl("~/DesktopModules/Events/Scripts/Validation.js") +
                             "\" type=\"text/javascript\"></script>";
                if (!Page.ClientScript.IsClientScriptBlockRegistered(csname))
                {
                    Page.ClientScript.RegisterClientScriptBlock(cstype, csname, cstext, false);
                }

                ddlLocations.EmptyMessage = Localization.GetString("NoLocations", LocalResourceFile);
                ddlLocations.Localization.AllItemsCheckedString =
                    Localization.GetString("AllLocations", LocalResourceFile);
                ddlLocations.Localization.CheckAllString =
                    Localization.GetString("SelectAllLocations", LocalResourceFile);
                if (Settings.Enablelocations == EventModuleSettings.DisplayLocations.SingleSelect)
                {
                    ddlLocations.CheckBoxes = false;
                }

                if (!Page.IsPostBack)
                {
                    //Bind DDL
                    var ctrlEventLocations = new EventLocationController();
                    var lstLocations = ctrlEventLocations.EventsLocationList(PortalId);

                    var arrLocations = new ArrayList();
                    if (Settings.Restrictlocations)
                    {
                        foreach (EventLocationInfo dbLocation in lstLocations)
                        {
                            foreach (int location in Settings.ModuleLocationIDs)
                            {
                                if (dbLocation.Location == location)
                                {
                                    arrLocations.Add(dbLocation);
                                }
                            }
                        }
                    }
                    else
                    {
                        arrLocations.AddRange(lstLocations);
                    }

                    if (lstLocations.Count == 0)
                    {
                        Visible = false;
                        SelectedLocation.Clear();
                        return;
                    }

                    //Restrict locations by events in time frame.
                    if (Settings.RestrictLocationsToTimeFrame)
                    {
                        //Only for list view.
                        var whichView = string.Empty;
                        if (!(Request.QueryString["mctl"] == null) && ModuleId ==
                            Convert.ToInt32(Request.QueryString["ModuleID"]))
                        {
                            if (Request["mctl"].EndsWith(".ascx"))
                            {
                                whichView = Request["mctl"];
                            }
                            else
                            {
                                whichView = Request["mctl"] + ".ascx";
                            }
                        }
                        if (whichView.Length == 0)
                        {
                            if (!ReferenceEquals(
                                    Request.Cookies.Get("DNNEvents" + Convert.ToString(ModuleId)), null))
                            {
                                whichView = Request.Cookies.Get("DNNEvents" + Convert.ToString(ModuleId)).Value;
                            }
                            else
                            {
                                whichView = Settings.DefaultView;
                            }
                        }

                        if (whichView == "EventList.ascx" || whichView == "EventRpt.ascx")
                        {
                            var objEventInfoHelper =
                                new EventInfoHelper(ModuleId, TabId, PortalId, Settings);
                            var lstEvents = default(ArrayList);

                            var getSubEvents = Settings.MasterEvent;
                            var numDays = Settings.EventsListEventDays;
                            var displayDate = default(DateTime);
                            var startDate = default(DateTime);
                            var endDate = default(DateTime);
                            if (Settings.ListViewUseTime)
                            {
                                displayDate = DisplayNow();
                            }
                            else
                            {
                                displayDate = DisplayNow().Date;
                            }
                            if (Settings.EventsListSelectType == "DAYS")
                            {
                                startDate = displayDate.AddDays(Settings.EventsListBeforeDays * -1);
                                endDate = displayDate.AddDays(Settings.EventsListAfterDays * 1);
                            }
                            else
                            {
                                startDate = displayDate;
                                endDate = displayDate.AddDays(numDays);
                            }

                            lstEvents = objEventInfoHelper.GetEvents(startDate, endDate, getSubEvents,
                                                                     new ArrayList(Convert.ToInt32(new[] {"-1"})),
                                                                     new ArrayList(Convert.ToInt32(new[] {"-1"})), -1,
                                                                     -1);

                            var eventLocationIds = new ArrayList();
                            foreach (EventInfo lstEvent in lstEvents)
                            {
                                eventLocationIds.Add(lstEvent.Location);
                            }
                            foreach (EventLocationInfo lstLocation in lstLocations)
                            {
                                if (!eventLocationIds.Contains(lstLocation.Location))
                                {
                                    arrLocations.Remove(lstLocation);
                                }
                            }
                        }
                    }

                    //Bind locations.
                    ddlLocations.DataSource = arrLocations;
                    ddlLocations.DataBind();

                    if (Settings.Enablelocations == EventModuleSettings.DisplayLocations.SingleSelect)
                    {
                        ddlLocations.Items.Insert(
                            0,
                            new RadComboBoxItem(Localization.GetString("AllLocations", LocalResourceFile),
                                                "-1"));
                        ddlLocations.SelectedIndex = 0;
                    }
                    ddlLocations.OnClientDropDownClosed =
                        "function() { btnUpdateClick('" + btnUpdate.UniqueID + "','" + ddlLocations.ClientID +
                        "');}";
                    ddlLocations.OnClientLoad = "function() { storeText('" + ddlLocations.ClientID + "');}";
                    if (Settings.Enablelocations == EventModuleSettings.DisplayLocations.SingleSelect)
                    {
                        foreach (int location in SelectedLocation)
                        {
                            ddlLocations.SelectedIndex =
                                ddlLocations.FindItemByValue(location.ToString()).Index;
                            break;
                        }
                    }
                    else
                    {
                        foreach (int location in SelectedLocation)
                        {
                            foreach (RadComboBoxItem item in ddlLocations.Items)
                            {
                                if (item.Value == location.ToString())
                                {
                                    item.Checked = true;
                                }
                            }
                        }

                        if (Convert.ToInt32(SelectedLocation[0]) == -1)
                        {
                            foreach (RadComboBoxItem item in ddlLocations.Items)
                            {
                                item.Checked = true;
                            }
                        }
                    }
                }
            }
            catch (Exception)
            {
                //ProcessModuleLoadException(Me, exc)
            }
        }

        public void btnUpdate_Click(object sender, EventArgs e)
        {
            StoreLocations();

            // Fire the LocationSelected event...
            var args = new CommandEventArgs(SelectedLocation.ToString(), null);
            if (LocationSelectedChangedEvent != null)
            {
                LocationSelectedChangedEvent(this, args);
            }
        }

        public delegate void LocationSelectedChangedEventHandler(object sender, CommandEventArgs e);

        private LocationSelectedChangedEventHandler LocationSelectedChangedEvent;

        public event LocationSelectedChangedEventHandler LocationSelectedChanged
        {
            add
                {
                    LocationSelectedChangedEvent =
                        (LocationSelectedChangedEventHandler) Delegate.Combine(
                            LocationSelectedChangedEvent, value);
                }
            remove
                {
                    LocationSelectedChangedEvent =
                        (LocationSelectedChangedEventHandler) Delegate.Remove(LocationSelectedChangedEvent, value);
                }
        }

        #endregion

        #region  Web Form Designer Generated Code

        //This call is required by the Web Form Designer.
        [DebuggerStepThrough]
        private void InitializeComponent()
        { }

        private void Page_Init(object sender, EventArgs e)
        {
            //CODEGEN: This method call is required by the Web Form Designer
            //Do not modify it using the code editor.
            InitializeComponent();
        }

        #endregion
    }
}